libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -1927,11 +1927,16 @@ bool RCOutput::set_serial_led_num_LEDs(const uint16_t chan, uint8_t num_leds, ou
 
     // we must hold the LED mutex while resizing the array
     WITH_SEMAPHORE(grp->serial_led_mutex);
-    // if already allocated then return
-    if (grp->serial_nleds > 0 || num_leds == 0
+    // if already allocated a different size then return
+    if ((grp->serial_nleds > 0 && num_leds != grp->serial_nleds)
          || (mode != MODE_NEOPIXEL && mode != MODE_PROFILED)) {
         return false;
     }
+    // if already allocated this channel
+    if (grp->serial_nleds > 0 && num_leds == grp->serial_nleds
```

Comment:
```suggestion
    if (grp->serial_nleds > 0 && num_leds <= grp->serial_nleds
```
Should be possible to use and smaller length too.

Reply:
This is already caught by the if above